### PR TITLE
Speed up H5/N5 writes

### PR DESCRIPTION
**Pull request by zacsimile**
Addresses #543. Fewer `if` statements and better chunking.

### CODE REVIEW

**Comment by AdvancedImagingUTSW** [comment]
Quick test on my computer. With `develop`...
image

With the PR...
image

So, at least on my personal computer, in a synthetic mode, not a massive decrease. But I could imagine this being a pretty OS and computer-specific thing. If it worked on an acquisition rig, that is what matters. Will review the edits now.

File: src/aslm/model/data_sources/bdv_data_source.py [hunk 1]
```diff
@@ -117,34 +123,26 @@ def set_metadata_from_configuration_experiment(
     def write(self, data: npt.ArrayLike, **kw) -> None:
         self.mode = "w"
```
Comment (AdvancedImagingUTSW) [inline src/aslm/model/data_sources/bdv_data_source.py:124]: Some quick numpydoc may not hurt. For example, to define what information is being passed in the **kw dictionary.

**Review by AdvancedImagingUTSW — APPROVED** [review]

**Comment by zacsimile** [comment]
How did you benchmark it? I got a 50x speed increase for a full stack, as shown in #543. The key isn't that each `write` call's internal speed is decreased, but that the larger chunk reduces the total number of calls the `zarr` library makes to save the N5. This is shown in the `cProfile` output, where I've included the number of calls and execution time for the top 10 most expensive functions in the stack saving process.

**Comment by zacsimile** [comment]
I haven't tried this out on the actual machine yet. This was benchmarked using cProfile on the saving tests.

**Comment by AdvancedImagingUTSW** [comment]
I only wrapped the write function in cdv_data_source with a timer. Could have not been the best place to benchmark it.

**Comment by zacsimile** [comment]
That should work, unless the write calls in `zarr` are non-blocking. For me, there is a distinct difference in write time using the `aslm` `@function_timer` decorator on the `bdv_data_source.py` write call. Seems to suggest they are blocking and the speedup is working with this benchmarking approach, too!

Old:

```
Conditions are multiposition: False per_stack: True z_stack: True stop_early: False
x: 1024 y: 2048 z: 3 c: 3 t: 2 positions: 1 per_stack: True
Function 'write' executed in 2.0677s
Function 'write' executed in 1.1498s
Function 'write' executed in 1.0584s
Function 'write' executed in 1.4082s
Function 'write' executed in 1.4349s
Function 'write' executed in 1.1758s
Function 'write' executed in 1.2580s
Function 'write' executed in 1.2001s
Function 'write' executed in 1.1910s
Function 'write' executed in 1.4105s
Function 'write' executed in 1.6153s
Function 'write' executed in 1.5295s
Function 'write' executed in 1.4605s
Function 'write' executed in 1.0740s
Function 'write' executed in 1.1127s
Function 'write' executed in 1.4157s
Function 'write' executed in 1.0710s
Function 'write' executed in 1.0610s
```

New:

```
Conditions are multiposition: False per_stack: True z_stack: True stop_early: False
x: 1024 y: 2048 z: 3 c: 3 t: 2 positions: 1 per_stack: True
Function 'write' executed in 0.4156s
Function 'write' executed in 0.0235s
Function 'write' executed in 0.0180s
Function 'write' executed in 0.0246s
Function 'write' executed in 0.0168s
Function 'write' executed in 0.0181s
Function 'write' executed in 0.0148s
Function 'write' executed in 0.0159s
Function 'write' executed in 0.0130s
Function 'write' executed in 0.0129s
Function 'write' executed in 0.0149s
Function 'write' executed in 0.0119s
Function 'write' executed in 0.0119s
Function 'write' executed in 0.0202s
Function 'write' executed in 0.0125s
Function 'write' executed in 0.0170s
Function 'write' executed in 0.0121s
Function 'write' executed in 0.0164s
```

**Pull request merged**